### PR TITLE
✅ Add safeCastTo16

### PR DESCRIPTION
**Pull request by robertabbott**
## Description
Adds safeCastTo16

## Checklist
- [x] Ran `forge snapshot`?
- [x] Ran `npm run lint`?
- [x] Ran `forge test`?

### CODE REVIEW

File: src/test/SafeCastLib.t.sol [hunk 1]
```diff
@@ -200,6 +205,12 @@ contract SafeCastLibTest is DSTestPlus {
         SafeCastLib.safeCastTo24(x);
     }
 
+    function testFailSafeCastTo16(uint256 x) public {
```
Comment (transmissions11) [inline src/test/SafeCastLib.t.sol:208]: missing a `testSafeCastTo16`

Comment (robertabbott) [reply src/test/SafeCastLib.t.sol:208]: https://github.com/transmissions11/solmate/pull/311/files#diff-1cf1e7c9ff842ab06001df49e0dae514a7141c6626a24ca4f5cafd887b4d4d09R54 ? or am i an idiot and completely misunderstanding

Comment (transmissions11) [reply src/test/SafeCastLib.t.sol:208]: you've got a concrete success test, but need a successful fuzz test too hehe

Comment (robertabbott) [reply src/test/SafeCastLib.t.sol:218]: i hate when i'm an idiot - added the test

**Pull request merged**